pkg/scheduler/framework/plugins/dynamicresources/dynamicresources.go

```diff
@@ -492,7 +497,34 @@ func (pl *dynamicResources) isSchedulableAfterClaimChange(logger klog.Logger, po
 		return framework.QueueSkip, nil
 	}
 
-	logger.V(4).Info("status of claim for pod got updated", "pod", klog.KObj(pod), "claim", klog.KObj(modifiedClaim))
+	logger.V(5).Info("status of claim for pod got updated", "pod", klog.KObj(pod), "claim", klog.KObj(modifiedClaim))
+	return framework.Queue, nil
+}
+
+// isSchedulableAfterPodChange is invoked for update pod events reported by
+// an informer. It checks whether that change adds the ResourceClaim(s) that the
+// pod has been waiting for.
+func (pl *dynamicResources) isSchedulableAfterPodChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	_, modifiedPod, err := schedutil.As[*v1.Pod](nil, newObj)
+	if err != nil {
+		// Shouldn't happen.
+		return framework.Queue, fmt.Errorf("unexpected object in isSchedulableAfterClaimChange: %w", err)
+	}
+
+	if pod.UID != modifiedPod.UID {
+		logger.V(7).Info("pod is not schedulable after change in other pod", "pod", klog.KObj(pod), "modifiedPod", klog.KObj(modifiedPod))
+		return framework.QueueSkip, nil
+	}
+
+	if err := pl.foreachPodResourceClaim(modifiedPod, nil); err != nil {
+		// This is not an unexpected error: we know that
+		// foreachPodResourceClaim only returns errors for "not
+		// schedulable".
+		logger.V(6).Info("pod is not schedulable after being updated", "pod", klog.KObj(pod))
+		return framework.QueueSkip, nil
+	}
```

Comment:
The alternative would be to just check whether the generated resource claims has shrunk down and enable scheduling, but I suspect this was a conscious decision.

Reply:
Actually shrunk down or changed, as in theory they claims can change as well. So not sure what is the most common case and whether the additional complication is worth the effort, but changes that are append only, won't infer the scheduling I think.

Reply:
This hint function only gets called if the code over in `events.go` has already determined that the pod's generated claims changed, so there is no need to check that again.

What the code here checks is "are all required claims generated".

Reply:
> This hint function only gets called if the code over in `events.go` has already determined that the pod's generated claims changed, so there is no need to check that again.

For instance instead of introducing a generic change event UpdatePodGeneratedResourceClaim, the alternative one could be scoped down (emitted) to changes that shank down the Pod's claims (if any of them disappeared - PodGeneratedResourceClaimEliminated) as this is the only case worth requeuing.

Unfortunately I don't know how often it happens that new ResourceClaims just grow, as this is the only case this approach could eliminate (isSchedulableAfterPodChange would not be called).

Reply:
> the alternative one could be scoped down (emitted) to changes that shank down the Pod's claims (if any of them disappeared - PodGeneratedResourceClaimEliminated) as this is the only case worth requeuing

It's the opposite: *adding* entries may make a pod schedulable because the ResourceClaim that it needs got added.

In practice, entries only get added, never removed nor updated. I don't think it's worth making the code more complex for situations that don't occur.

Reply:
Ah right, claims gets fulfilled over time. There's indeed no place for optimization.
Sorry for the dumb question, just ramping up and connecting dots slowly :)